doc/ota_updates/ota_updates.md

```diff
@@ -61,75 +81,175 @@ Uploading modules wirelessly from Arduino IDE is intended for the following typi
 
- ![selection og OTA port](ota-ide-ota-port-selection.png)
+  ![selection og OTA port](ota-ide-ota-port-selection.png)
 
 5. Now get ready for your first OTA upload by changing configuration settings as follows:
 
- ![configuration of OTA upload](ota-ide-ota-upload-configuration.png)
+  ![configuration of OTA upload](ota-ide-ota-upload-configuration.png)
 
- **Note:** If you do not see “Upload Using: OTA” option available for “NodeMCU 1.0 (ESP-12E Module)” board, please upload the latest [boards.txt](https://github.com/esp8266/Arduino/blob/master/boards.txt) file from Github repository, replace existing file and restart Arduino IDE.
+  **Note:** If you do not see “Upload Using: OTA” option available for “NodeMCU 1.0 (ESP-12E Module)” board, please upload the latest [boards.txt](https://github.com/esp8266/Arduino/blob/master/boards.txt) file from GitHub repository, replace existing file and restart Arduino IDE.
 
 6. If you have successfully completed all the above steps, you can upload (Ctrl+U) the same (or any other) sketch over OTA:
 
- ![OTA upload complete](ota-ide-ota-upload-complete.png)
+  ![OTA upload complete](ota-ide-ota-upload-complete.png)
 
 **Note** To be able to upload your sketch over and over again using OTA, you need to embed OTA routines inside. Please use DNS_SD_Arduino_OTA.ino as an example.
 
-##### ArduinoOTA Configuration
 
-1. Get the following software:
- - Arduino-PR-4107-BUILD-421 - https://github.com/esp8266/Arduino/pull/984#issuecomment-155905800
- - Latest git version of pacakge - https://github.com/esp8266/Arduino#using-git-version-
- - Python 2.7.10
+#### ArduinoOTA
 
-2. Proceed to step 2 under [Classic OTA Configuration](#classic-ota-configuration) using BasicOTA.ino or OTALeds.ino skech instead.
+1. Upload and install the following software:
+  - Arduino-PR-4107-BUILD-421 - https://github.com/esp8266/Arduino/pull/984#issuecomment-155905800
+  - Latest git version of platform package - https://github.com/esp8266/Arduino#using-git-version-
+  - Python 2.7
+
+2. Proceed to step 2 under [Classic OTA Configuration](#classic-ota-configuration) using BasicOTA.ino or OTALeds.ino sketch instead.
 
 3. Carry on with remaining steps.
 
 
+## Web Browser
+
+Updates described in this chapter are done with a web browser that can be useful in the following typical scenarios:
+- after application deployment if loading directly from Arduino IDE is inconvenient or not possible
+- after deployment if user is unable to expose module for OTA from external update server
+- to provide updates after deployment to small quantity of modules when setting an update server is not practicable
+
+
+### Requirements
+
+- The ESP and the computer must be connected to the same network.
+
+
+### Implementation Overview
+
+Updates with a web browswer are implemented using ```ESP8266HTTPUpdateServer``` class together with ```ESP8266WebServer``` and ```ESP8266mDNS``` classes. The following code is required to get it work:
```

Comment:
Thanks, Christopher
Will correct in next pull request
Regards, Krzysztof

BTW - [your Murmur Study](http://christopherbaker.net/projects/murmur-study/) is awesome :+1: